crates/router/src/core/payments/helpers.rs

```diff
@@ -900,22 +917,32 @@ pub fn validate_amount_to_capture_and_capture_method(
             .map(|payment_attempt| payment_attempt.capture_method.unwrap_or_default()))
         .unwrap_or_default();
     if capture_method == api_enums::CaptureMethod::Automatic {
-        let original_amount = request
-            .amount
-            .map(MinorUnit::from)
-            .or(payment_attempt.map(|payment_attempt| payment_attempt.amount));
+        let original_amount =
+            request.amount.map(MinorUnit::from).or(payment_attempt
+                .map(|payment_attempt| payment_attempt.net_amount.get_order_amount()));
         let surcharge_amount = request
             .surcharge_details
             .map(|surcharge_details| surcharge_details.get_total_surcharge_amount())
             .or_else(|| {
                 payment_attempt.map(|payment_attempt| {
-                    payment_attempt.surcharge_amount.unwrap_or_default()
-                        + payment_attempt.tax_amount.unwrap_or_default()
+                    payment_attempt
+                        .net_amount
+                        .get_surcharge_amount()
+                        .unwrap_or_default()
+                        + payment_attempt
+                            .net_amount
+                            .get_tax_on_surcharge()
+                            .unwrap_or_default()
                 })
             })
             .unwrap_or_default();
-        let total_capturable_amount =
-            original_amount.map(|original_amount| original_amount + surcharge_amount);
+        let shipping_cost = request
+            .shipping_cost
+            .or(payment_attempt
+                .and_then(|payment_attempt| payment_attempt.net_amount.get_shipping_cost()))
+            .unwrap_or_default();
+        let total_capturable_amount = original_amount
```

Comment:
Why we are calculating this instead of using `get_total_amount`?

Reply:
This is being called in validate_request flow also, at that point payment_attempt is not present yet. So, we have to take details from request and calculate the amount